packages/expo-modules-core/ios/JSI/EXJavaScriptObject.h

```diff
@@ -91,5 +91,6 @@ NS_SWIFT_NAME(JavaScriptObject)
 #pragma mark - Deallocator
 
 - (void)setObjectDeallocator:(void (^ _Nonnull)(void))deallocatorBlock;
+- (void)triggerDeallocatorForTesting;
```

Comment:
I wonder if we can have a `TESTING` flag in our unit tests and then hide this method.

Reply:
personally, i am not a fan to have special UNIT_TEST flag to do that. but i did add one in expo-modules-core on android because of hermes/jsc linking. will try to study how to do that in cocoapods.

Reply:
But it's also not something that we want to publicly expose 😢 It shouldn't be a part of the public API and I believe there might be more cases like this.

Reply:
i got not luck to use a compiler flag for unit testing only.

to think the problem more, i feel like the current `triggerDeallocatorForTesting()` approach is fundamentally same as the internal call to `SharedObjectRegistry.delete()` that we don't actually do GC for the object but call their underlying deallocator.
so maybe the better way is just removing the test case
https://github.com/expo/expo/blob/d2ff888fda6ceb8b29b9747f6844b151c4dc219f/packages/expo-modules-core/ios/Tests/SharedObjectRegistrySpec.swift#L73-L78

will update the pr for this direction and see how do you feel about that.

Reply:
Ok, I'm fine with just removing the test 🤷‍♂️ Better than nothing 😅